src/utils/index.ts

```diff
@@ -100,3 +100,20 @@ export function getDrives(...paths: string[]): string[] {
   const rx = new RegExp('^[a-zA-Z]:');
   return paths.map(x => (rx.exec(x) || [])[0]);
 }
+
+export function flatten(object: object, separator = '.'): object {
+  const isValidObject = (value): boolean => {
+    if (!value) { return false; }
+    const isArray = Array.isArray(value);
+    const isBuffer = Buffer.isBuffer(value);
+    const isΟbject = Object.prototype.toString.call(value) === "[object Object]";
+    const hasKeys = !!Object.keys(value).length;
+    return !isArray && !isBuffer && isΟbject && hasKeys;
+  };
+  return Object.assign({}, ...function _flatten(child, path = []) {
+    return [].concat(...Object.keys(child)
+      .map(key => isValidObject(child[key])
+        ? _flatten(child[key], path.concat([key]))
+        : { [path.concat([key]).join(separator)]: child[key] }));
+  }(object));
+}
```

Comment:
Even at risk of being less terse, I don't really like using a named function expression recursively using an IIFE with spread operators. I think that, although correct, this syntax is really convoluted and something like this will improve readability:

```ts
  const _flatten = (child, path = []): object[] => {
    return [].concat(...Object.keys(child)
      .map(key => isValidObject(child[key])
        ? _flatten(child[key], [...path, key])
        : { [[...path, key].join(separator)]: child[key] }));
  };

  return Object.assign({}, ..._flatten(obj));

```

Note that I've modified some of the `[].concat` with `spread operators` and only left the strictly needed ones.

Another suggestion would be to change the `object parameter` with `obj` or something similar to avoid confusion with the `object` ts keyword.

Finally, I would also add the return type annotation to the `_flatten` function in order to make it more clear.